modules/saku-policy-store/src/saku/routes.clj

```diff
@@ -0,0 +1,234 @@
+        db (dal/db dal-obj)
+        ctx {:dal-obj dal-obj :db db}
+        resource-policies (mapv ->evaluate-policy (dal/get-policies ctx {:drns drns :policyType "Resource"}))
+        drn->resource-policy (into {} (map (juxt :drn identity)) resource-policies)
+        identity-policies (mapv ->evaluate-policy (dal/get-policies ctx {:drns identities :policyType "Identity"}))
+        drn->identity-policy (into {} (map (juxt :drn identity)) identity-policies)]
+    (into []
+      (map (fn [drn]
+             (let [resource-policy (drn->resource-policy drn)
+                   evaluate-argm {:drn                   drn
+                                  :action                actionId
+                                  :resource-policy       resource-policy
+                                  :identity-policies-map drn->identity-policy}
+                   result (core/evaluate-one evaluate-argm)]
+               (log/info (assoc evaluate-argm
+                           :result result
+                           :msg "Evaluating effect..."))
+               {:drn drn :result result})))
+      drns)))
+
+(defn evaluate-one-handler
+  [request]
+  (let [{:saku/keys [ctx]
+         :keys      [parameters]} request
+        many-result (evaluate-many {:dal-obj (:dal-obj ctx)}
+                      (-> (:body parameters)
+                        (update :drn vector)
+                        (sets/rename-keys {:drn :drns})))
+        result (-> many-result first :result)]
+    {:status 200
+     :body   result}))
+
+(defn evaluate-many-handler
+  [request]
+  (let [{:saku/keys [ctx]
+         :keys      [parameters]} request
+        result (evaluate-many {:dal-obj (:dal-obj ctx)} (:body parameters))]
+    {:status 200
+     :body   result}))
+
+(def wrap-ctx
+  {:name        ::ctx
+   :description "Add ctx to request map."
+   :wrap        (fn [handler ctx]
+                  (fn [request]
+                    (handler (assoc request :saku/ctx ctx))))})
+
+(def wrap-request-id
+  {:name        ::request-id
+   :description "Add request-id to the log mdc"
+   :wrap        (fn [handler]
+                  (fn [request]
+                    (let [request-id (str (random-uuid))]
+                      (log/with-context+ {:request_id request-id}
+                        (log/info {:msg            "Received request"
+                                   :uri            (:uri request)
+                                   :request-method (:request-method request)
+                                   :host           (get-in request [:headers "host"])
+                                   :user-agent     (get-in request [:headers "user-agent"])})
+                        (let [response (handler request)]
+                          (log/info {:msg    "Finished request"
+                                     :status (:status response)})
+                          (assoc-in response [:headers "x-request-id"] request-id))))))})
```

Comment:
This is a very handy way to debug calls to the API. Every incoming request will have an ID associated with it. Within the request execution, all log statements will include the request ID. This will allow us to observe all logs for a single API operation. We also include the request ID in the response headers, which is used in the client's debug mode.